apps/web/app/modules/Admin/EditCourse/CourseLessons/NewLesson/QuizLessonForm/QuizLessonForm.tsx

```diff
@@ -105,20 +106,52 @@ const QuizLessonForm = ({
   const addQuestion = useCallback(
     (questionType: string) => {
       const questions = form.getValues("questions") || [];
-
+  
+      const getOptionsForQuestionType = (type: string): QuestionOption[] => {
+        const singleChoiceTypes = [
+          QuestionType.SINGLE_CHOICE,
+          QuestionType.MULTIPLE_CHOICE,
+          QuestionType.MATCH_WORDS,
+          QuestionType.PHOTO_QUESTION,
+        ];
+  
+        const noOptionsRequiredTypes = [
+          QuestionType.FILL_IN_THE_BLANKS_TEXT,
+          QuestionType.FILL_IN_THE_BLANKS_DND,
+          QuestionType.BRIEF_RESPONSE,
+          QuestionType.DETAILED_RESPONSE,
+        ];
+  
+        if (singleChoiceTypes.includes(type as QuestionType)) {
```

Comment:
will it work if you change all `type: string` and `questionType: string` (related to this function) to the `QuestionType` type you used in type cast? I think it might be simpler to debug in future without this `type as QuestionType`